src/runtime/module_jit_ref_count.cpp

```diff
@@ -0,0 +1,29 @@
+#include "./runtime_internal.h"
+
+extern "C" {
+
+WEAK void *halide_jit_module_argument = NULL;
+WEAK void (*halide_jit_module_adjust_ref_count)(void *arg, int32_t count) = NULL;
+
+}
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+WEAK void halide_use_jit_module() {
```

Comment:
Please see the code I just pushed to gpu_api_naming. This moves the ref count manipulations into the device interface layer and provides a way to increment and decrement the ref count through the device interface itself. If the device_interface.cpp code is part of libHalide.a, it will not be freed when the refcount goes to 0. I'll try to figure out the build logic around making device_interface.cpp part of libHalide shortly. This change also raises the question of how to handle exceptions in the runtime as I believe the goal is to allow client code to throw an exception from a provided handler in the case of an error.